pkg/apis/kf/v1alpha1/app_lifecycle.go

```diff
@@ -110,6 +118,40 @@ func (status *AppStatus) PropagateEnvVarSecretStatus(secret *v1.Secret) {
 	status.manage().MarkTrue(AppConditionEnvVarSecretReady)
 }
 
+// PropagateServiceBindingsStatus updates the service binding readiness status.
+func (status *AppStatus) PropagateServiceBindingsStatus(bindings []servicecatalogv1beta1.ServiceBinding) {
+
+	var bindingNames []string
+	for _, binding := range bindings {
+		bindingNames = append(bindingNames, binding.Name)
+	}
+	status.ServiceBindingNames = bindingNames
```

Comment:
WDYT about creating an additional conditions array like `ServiceBindingConditions` so each can be reflected individually?

You might be able to re-use knative's manager logic with something like that too to get an overall status

Reply:
I added the ServiceBindingConditions but I did not create a 2nd manager.